data/data/gcp/cluster/main.tf

```diff
@@ -107,3 +108,20 @@ resource "google_compute_image" "cluster" {
 
   licenses = var.gcp_image_licenses
 }
+
+data "google_compute_image" "cluster_images" {
+  count = var.gcp_preexisting_image ? 0 : 1
+  name  = "${var.cluster_id}-rhcos-image"
+
+  depends_on = [google_compute_image.cluster]
+}
+
+resource "google_tags_tag_binding" "image_tag_binding" {
+  for_each = { for key, val in var.gcp_extra_tags : key => val if ! var.gcp_preexisting_image }
+
+  parent = format("//compute.googleapis.com/projects/%s/global/images/%s",
+    var.gcp_project_id,
+    data.google_compute_image.cluster_images[0].image_id,
```

Comment:
Instead of using the data source here, you can just read from the resource, it would be something like:

```
google_compute_image.cluster[0].image_id
```

Reply:
Yeah, but google_compute_image API is not exporting image_id attribute. Hence had to make use of data source to fetch the image_id value. [Link](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_image.html#attributes-reference) has details on the list of attributes that are made available by the API.